rfc-163-cdn-rationalisation.md

```diff
@@ -0,0 +1,85 @@
+
+The lack of feature parity between the two CDNs presents us with several issues:
+
+- Lack of clarity or confidence around what works in Cloudfront (e.g. developers have to be aware of this when building things, SMT need to be reminded of impact failover will have, we may be more hesitant in triggering the failover, etc.)
+- Complexities also introduced downstream (e.g. if we failover and A/B testing stops working, data analysts now need to scrub out the failover period in their analysis)
+- We're less likely to drill the failover in Production to ensure it works (as it will cause a somewhat degraded experience)
+- It restricts our ability to consider a multi CDN strategy where traffic is split between two CDNs simultaneously 
+
+A number of the things that we currently handle at the CDN level might perhaps be better handled at other places in the stack. By moving things out of the CDN layer, we minimise the amount of duplicated effort in maintaining equivalent but different configurations for each CDN.
+
+Fastly are recommending the use of [Compute@Edge](https://www.fastly.com/products/edge-compute) for future development on their platform. Using Compute@Edge would allow us to express our CDN logic in a programming language, using an SDK provided by Fastly, instead of having to grapple with VCL. This could make it easier to maintain the functionality that can't be relocated to other parts of the stack, as well as allowing us to set up integration tests for our CDN logic.
+
+Across GDS we are standardising on infrastructure as code, and the use of Terraform to describe this infrastructure. Our Fastly service configuration has so far avoided this treatment, and is still deployed using custom Ruby code. We should address this by replacing our [handwritten Ruby scripts](https://github.com/alphagov/govuk-cdn-config/tree/main/lib) in [govuk-cdn-config](https://github.com/alphagov/govuk-cdn-config) with Terraform.
+
+## Proposal
+
+### Where possible, relocate logic from the CDN layer to other parts of the stack
+
+Please see the [appendix](rfc-163/relocating-logic-from-cdn.md) for specific examples of what functionality we are proposing to move, and where we are proposing to move it to. Note however that this is not an exhaustive list, and is subject to change; the main point is that we want to move as much functionality out of the CDN as possible, to make it easier to maintain.
+
+### Migrate our Fastly VCL to Compute@Edge
+
+Compute@Edge is Fastly's edge compute product, and their recommended approach for developing services moving forwards.
```

Comment:
I seem to remember reading something along these lines, but I can’t find a web page to back that up - can you provide a link if possible?

Reply:
I suppose "their recommended approach" may be an exaggeration - their dev docs homepage points you towards Compute@Edge as opposed to VCL now, but I think both approaches are still supported. Will rephrase this!